pkg/controller/seed-controller-manager/addon/addon_controller.go

```diff
@@ -264,6 +263,21 @@ func (r *Reconciler) reconcile(ctx context.Context, log *zap.SugaredLogger, addo
 		return reqeueAfter, nil
 	}
 
+	// This handles manifests that should be removed manually, because kubectl --prune does not always delete obsolete resources.
```

Comment:
Please explain more why this is here. Is kubectl broken? What is this "not always"?

/lgtm cancel